src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7845,25 +7841,25 @@ public int GetHashCode((NullabilityInfo info, TypeSymbol type) obj)
             }
         }
 
-        private sealed class ExpressionAndSymbolEqualityComparer : IEqualityComparer<(BoundNode expr, Symbol sym)>
+        private sealed class ExpressionAndSymbolEqualityComparer : IEqualityComparer<(BoundNode? expr, Symbol sym)>
         {
             internal static readonly ExpressionAndSymbolEqualityComparer Instance = new ExpressionAndSymbolEqualityComparer();
 
             private ExpressionAndSymbolEqualityComparer() { }
 
-            public bool Equals((BoundNode expr, Symbol sym) x, (BoundNode expr, Symbol sym) y)
+            public bool Equals((BoundNode? expr, Symbol sym) x, (BoundNode? expr, Symbol sym) y)
             {
-                Debug.Assert(x is ({ }, { }) && y is ({ }, { }));
+                Debug.Assert(x.sym is object && y.sym is object);
 
                 // We specifically use reference equality for the symbols here because the BoundNode should be immutable.
                 // We should be storing and retrieving the exact same instance of the symbol, not just an "equivalent"
                 // symbol.
                 return x.expr == y.expr && (object)x.sym == y.sym;
             }
 
-            public int GetHashCode((BoundNode expr, Symbol sym) obj)
+            public int GetHashCode((BoundNode? expr, Symbol sym) obj)
             {
-                Debug.Assert(obj is ({ }, { }));
+                Debug.Assert(obj.sym is object);
                 return Hash.Combine(obj.expr, obj.sym.GetHashCode());
```

Comment:
>obj.expr [](start = 36, length = 8)

Is it Ok to pass null here?

Reply:
Yes, it's appropriately nullchecked. Sam's PR annotates Hash.Combine here: [https://github.com/dotnet/roslyn/pull/38882/files#diff-a4f01f0c5d4e289dfae2bc84ec50366cR33](https://github.com/dotnet/roslyn/pull/38882/files#diff-a4f01f0c5d4e289dfae2bc84ec50366cR33)

---
In reply to: [329736470](https://github.com/dotnet/roslyn/pull/38769#discussion_r329736470) [](ancestors = 329736470)